### DISCUSSION TITLE
Webworker and style loader - causing load issues with babylon js inspector

### DISCUSSION THREAD

**Discussion by rdpeake** [webpack/webpack#15918] [Q&A]
This is likely a non-standard use case - however the original disussion between myself and babylonjs devs is here: [js forums](https://forum.babylonjs.com/t/importing-inspector-in-typescript-cause-failure-if-engine-started-with-offscreen-canvas/19234/9)

The summary is that babylonJS has a component that can be imported via a simple `import 'script'` call, but that doesn't actually get enabled unless you then call an actual command there after. The code i am using is designed to be aware of if you are running normally, or if it is inside a webworker, but because the component is added via typescript import i'm fairly sure i cannot have runtime conditionals on the import statement.

When this component is added, the webpack style-loader component is run and it tries to access the DOM to inject various stylings which ultimately are not used. a minimal reproduction of this issue is here: [github project](https://github.com/rdpeake/babylon-inspector-repro/tree/master) though the project hasn't been updated the related issues persist with the latest versions.

my question is, is there a change that either babylonjs devs, or webpack devs can make such that style-loader is dom aware, or is there something i can edit to achieve this. my current solution is to patch styleDomAPI locally to be a noop if it has no document object.

**Comment by vankop** [comment]
please describe problem more clear... ( better with small reproducible example ) I didn't understand what is a problem.

**Reply by rdpeake** [reply]
i have a reproduction of the issue here: https://github.com/rdpeake/babylon-inspector-repro

The intention is that the project should run and show a ball in its initial state. It however does not, but can be patched (via npx patch-package) to implement the work around i have found that seems to work.

The packages that i patch to get it to work are not mine, but is you look at the forum post i linked above they point out that the parts i am patching are also not theirs (i am patching styleDomAPI from webpack style-loader ).

the other ways to get the repro to run are listed in the readme.


My question is - can babylonjs make a change themselves to get this project to run without needing to add my patch, or does style-loader need to change in order for this to work?
